### ISSUE TITLE
Add an hourly heatmap chart type

### ISSUE THREAD

**Issue by tomlut** [grafana/grafana#87947]
**Why is this needed**:

According to this [blog post](https://grafana.com/blog/2020/06/16/plugin-showcase-the-hourly-heatmap-panel-built-on-grafanas-new-plugin-platform/) the 3rd party carpet plot and its replacement, hourly heat map, have been extremely popular plug-ins. I personally use it extensively and find it invaluable for easy and quick data visualisation checks. 

Neither plug-ins are being actively supported and neither are compatible with the latest releases of Grafana. I have not updated for a while and am still running Grafana 10.1.2 because of this. 

The Grafana Labs Heatmap chart does not come close to the functionality these 3rd party plug-ins offered. Compare the information available:

Grafana Labs Heatmap
<img width="704" alt="temperature_heatmap" src="https://github.com/grafana/grafana/assets/10679300/55319ad1-06a3-479e-b184-cebfc0eadef1">

Carpet-plot:
![Screenshot 2024-05-16 at 09-50-28 Grafana – Home Assistant](https://github.com/grafana/grafana/assets/10679300/8b5e2f61-1e98-4458-820b-51794b4554ec)

**What would you like to be added**:

A chart that has time on the Y-axis, binned by hour, date on the X-axis, and measured value represented bu the colour of the cell. e.g.

![Screenshot 2024-05-16 at 09-48-29 Grafana – Home Assistant](https://github.com/grafana/grafana/assets/10679300/ebc3a45b-955b-433a-80e9-4b942bb59829)

**Who is this feature for?**

The large number of people who used to use the 3rd party plug-ins.

**Comment by hole222** [comment]
I still think this graph type is sorely needed.

**Comment by nioc** [comment]
I agree, I used carpet-plot and it was very useful for understanding daily data by hours.

**Comment by lowlydba** [comment]
I agree, this is a relatively simple but extremely useful chart type that I would love to be able to use on modern versions.
